app/services/sequence_update_service.rb

```diff
@@ -0,0 +1,20 @@
+class SequenceUpdateService
+  attr_accessor :workflow_id
+
+  def initialize(workflow_id)
+    self.workflow_id = workflow_id
+  end
+
+  def move(delta)
+    diff = case delta
+           when 'top'
+             -Float::INFINITY
+           when 'bottom'
+             Float::INFINITY
+           else
+             delta.to_i
+           end
+
+    Workflow.find(workflow_id).move_internal_sequence(diff)
```

Comment:
All of the logic that is contained in the model feels like it would belong here instead. Obviously, it still *works* this way, but to me, the name indicates that it actually does the update logic, so it seems odd that we would create this service simply to differentiate between top/bottom/integer.

There's a lot of logic being shoved into the model there, and I think it would be better encapsulated here.